### DISCUSSION TITLE
Avoiding Duplicate Message Consumption in Scaled Microservices

### DISCUSSION THREAD

**Discussion by MilynCodeCraft** [rabbitmq/rabbitmq-server#11246] [Questions] [answered]
Hey im Seeking advice on architecting a solution for an existing microservices environment where Service A publishes messages, and  Services B and C need to consume them. How can we ensure that only one instance of each service receives the message, avoiding duplication across multiple instances when we have scaled up ex Having 3 instances or replicas of service B and Service C , Here Service b and service c should receive the message but not the 2nd or 3rd instance /replica of service b and c ? Looking for insights and best practices from the community.

**Comment by lukebakken** [comment, marked as answer]
Please see the documentation:

https://www.rabbitmq.com/docs/consumers#priority

https://www.rabbitmq.com/docs/consumers#single-active-consumer

https://www.rabbitmq.com/tutorials#2-work-queues
